jwst/assign_wcs/niriss.py

```diff
@@ -1,26 +1,116 @@
 from __future__ import (absolute_import, unicode_literals, division,
                         print_function)
 
+import logging
+
 from asdf import AsdfFile
 from astropy import coordinates as coord
 from astropy import units as u
-
+from astropy.modeling.models import Const1D, Mapping
+from gwcs import wcs
 import gwcs.coordinate_frames as cf
-from . import pointing
 from .util import not_implemented_mode
-
+from ..transforms.models import NirissSOSSModel
 
 def create_pipeline(input_model, reference_files):
     '''
     get reference files from crds
 
     '''
+
     exp_type = input_model.meta.exposure.type.lower()
     pipeline = exp_type2transform[exp_type](input_model, reference_files)
 
     return pipeline
 
 
+def niriss_soss_set_input(model, order_number):
+    """
+    Get the right model given the order number.
+
+    Parameters
+    ----------
+    model - Input model
+    order_number - the, well, order number desired
+
+    Returns
+    -------
+    WCS - the WCS corresponding to the order_number
+
+    """
+
+    # Make sure the order number is correct.
+    if order_number < 1 or order_number > 3:
+        raise ValueError('Order must be between 1 and 3')
+
+    # Return the correct transform based on the order_number
+    obj = model.meta.wcs.forward_transform.get_model(order_number)
+
+    # use the size of the input subarray7
+    detector = cf.Frame2D(name='detector', axes_order=(0, 1), unit=(u.pix, u.pix))
+    sky = cf.CelestialFrame(reference_frame=coord.ICRS())
+    pipeline = [(detector, obj),
+                (sky, None)
+                ]
+
+    return wcs.WCS(pipeline)
+
+
+def soss_load_wavelength_lookup(reference_files):
+    """
+    Load the wavelength lookup from the configuration file;.
+
+    Parameters
+    ----------
+    reference_files - Currently should be just a single file.
+
+    Returns
+    -------
+    wavelength_lookup - Return the Tabular2D lookup model.
+
+    """
+    wavelength_lookup = AsdfFile.open(reference_files['distortion'])
+    return wavelength_lookup
+
+
+def niriss_soss(input_model, reference_files):
+    """
+    The NIRISS SOSS pipeline includes 3 coordinate frames -
+    detector, focal plane and sky
+
+    reference_files={'distortion': 'soss_wavelengths_configuration.asdf'}
+    """
+
+    # Get the target RA and DEC, they will be used for setting the WCS RA and DEC based on a conversation
+    # with Kevin Volk.
+    target_ra, target_dec = 0.0, 0.0
+    try:
+        target_ra = float(input_model['meta.target.ra'])
+        target_dec = float(input_model['meta.target.dec'])
+    except:
+        # There was an error getting the target RA and DEC, so we are not going to continue.
+        raise ValueError('Problem getting the TARG_RA or TARG_DEC from input model {}'.format(input_model))
+
+    # Define the frames
+    detector = cf.Frame2D(name='detector', axes_order=(0, 1), unit=(u.pix, u.pix))
+    sky = cf.CelestialFrame(reference_frame=coord.ICRS())
```

Comment:
I can add that but I looked at some of the other scripts (e.g., miri.py) and none of them have `axes_order`, `axes_names`, nor `units` for they sky